src/hooks/useCurrentRoundId.ts

```diff
@@ -12,16 +14,23 @@ const contract = createVoidSignerVotingContractInstance(
 
 // This can be accessed without logging the user in.
 export default function useCurrentRoundId() {
+  const { addError } = useContext(ErrorContext);
+
   const { data, error, isFetching, refetch } = useQuery<string>(
     "roundId",
     () => {
-      return queryCurrentRoundId(contract).then((res) => {
-        if (res) {
-          return res;
-        } else {
+      return queryCurrentRoundId(contract)
+        .then((res) => {
+          if (res) {
+            return res;
+          } else {
+            return "";
+          }
+        })
+        .catch((err) => {
+          addError(err.message);
           return "";
-        }
-      });
+        });
```

Comment:
i've mentioned this a while ago, but this general react-query pattern is used a lot in this code, but i really think it could be simplified if you require the view to always check if data exists before using it.  Most of this file could be squashed into a few lines:
```js
export default function useCurrentRoundId() {
  const { addError } = useContext(ErrorContext);

  return useQuery<string>(
    "roundId",
    () => {
      return queryCurrentRoundId(contract).catch(addError)  // addError is optional here, you could let this throw to pass error to view in some situations
    }
  );
}
```

The only changes are `addError` accepts a full error object rather than just a string and any view that consumes this hook must check that `data` exists before using it. what do you think about this?

Reply:
Would you suggest I pass in empty arrays in places it's looking for the data? Like if it's voteData, it's actually voteData || [] ?